src/core/contracts/utilities.ts

```diff
@@ -181,3 +186,148 @@ export const subscribeToEvent = async (
     provider.off(filter);
   };
 };
+
+/**
+ * BlockRangeOptions: represents parameters for specifying chain filter and block limits.
+ */
+export type BlockRangeOptions = {
+  filter: EventFilter;
+  walkbackBlocks: number;
+  latestBlock: number;
+  earliestBlock: number;
+};
+
+export const MAX_ITERATOR_WALKBACK_BLOCKS = 10000;
```

Comment:
I'm unsure about this constant even being here, however, I figured it would be good to set some kind of maximum, and  this is the same value as the current Ethereum log fetch limit.